module/Finna/src/Finna/Search/DateRangeFilterTrait.php

```diff
@@ -0,0 +1,197 @@
+     *
+     * @return array $filterQuery
+     */
+    public function getFilterSettings(): array
+    {
+        $result = parent::getFilterSettings();
+
+        // Special processing for date range filters
+        $dateRangeField = $this->getDateRangeSearchField();
+        if ($dateRangeField) {
+            foreach ($result as &$filter) {
+                $dateRange = strncmp(
+                    $filter,
+                    "$dateRangeField:",
+                    strlen($dateRangeField) + 1
+                ) == 0;
+                if ($dateRange) {
+                    [, $value] = $this->parseFilter($filter);
+                    [$op, $range] = explode('|', $value);
+                    $op = $op == 'within' ? 'Within' : 'Intersects';
+                    $filter = "{!field f=$dateRangeField op=$op}$range";
+                }
+            }
+        }
+        return $result;
+    }
+
+    /**
+     * Initialize date range filter (search_daterange_mv)
+     *
+     * @param \Laminas\Stdlib\Parameters $request Parameter object representing user
+     * request.
+     *
+     * @return void
+     */
+    public function initSpatialDateRangeFilter($request): void
+    {
+        $dateRangeField = $this->getDateRangeSearchField();
+        if (!$dateRangeField) {
+            return;
+        }
+        $type = $request->get("{$dateRangeField}_type");
+        if (!$type) {
+            // VuFind 1
+            $type = $request->get($this->spatialDateRangeFieldTypeVF1);
+        }
+        if (!$type) {
+            $type = $this->spatialDateRangeDefaultType;
+        }
+
+        $from = $to = null;
+        $found = false;
+        // Date range filter
+        if (($reqFilters = $request->get('filter')) && is_array($reqFilters)) {
+            foreach ($reqFilters as $f) {
+                [$field, ] = $this->parseFilter($f);
+                if (
+                    $field == $dateRangeField
+                    || $field == $this->spatialDateRangeFieldVF1
+                ) {
+                    if ($range = $this->parseDateRangeFilter($f)) {
+                        $from = $range['from'];
+                        $to = $range['to'];
+                        if (
+                            isset($range['type'])
+                            && $range['type'] !== $this->spatialDateRangeDefaultType
+                        ) {
+                            $type = $range['type'];
+                        }
+                        $found = true;
+                        break;
+                    }
+                }
+            }
+        }
+
+        // Uninitialized VuFind1 date range query
+        if (!$found && $request->get('sdaterange')) {
+            // Search for VuFind1 search_sdaterange_mvfrom, search_sdaterange_mvto
+            $from = $request->get('search_sdaterange_mvfrom');
+            $to = $request->get('search_sdaterange_mvto');
+            if (!empty($from) || !empty($to)) {
```

Comment:
Toimisko joku if ($from || $to) ja sit kattoa vaan if (!$from) ja if (!$to) alla